### PR TITLE
LSPR refactor

### PR DESCRIPTION
**Pull request by ncclementi**
This is a refactoring of the LSPR application.

 Changes (according discussion with @gforsyth ):

- Separate main function. We have one `main()` for biomolecular electrostatics located in `pygbe.main` and a separate one for LSPR applications located in `pygbe.lspr`

- Adapt tests to run accordingly the change mentioned above. 

- Automatize download of meshes from zenodo, required  to run convergence tests. 

**Note:** Regression and convergence test passed properly

### CODE REVIEW

File: tests/convergence_tests/check_for_meshes.py [hunk 1]
```diff
@@ -18,28 +18,44 @@ def download_zip_with_progress_bar(url):
                 f.write(chunk)
                 f.flush()
 
-def unzip(meshzip):
+def unzip(meshzip, folder_name, raname_folder):
```
Comment (gforsyth) [inline tests/convergence_tests/check_for_meshes.py]: *`rename`

File: tests/convergence_tests/check_for_meshes.py [hunk 2]
```diff
@@ -18,28 +18,44 @@ def download_zip_with_progress_bar(url):
...
+    '''Unzip a zip file
+
+    Arguments:
+    ----------
+    meshzip      : file we want to unzip
+    folder_name  : str, name of the folder after extracting.
+                  We want this to modify it when download, to follow pygbe 
+                  problem folder design for running.
+    raname_folder: str, name of the folder were we want to store the download.
```
Comment (gforsyth) [inline tests/convergence_tests/check_for_meshes.py]: again `rename`

**Comment by gforsyth** [comment]
Looks good, @ncclementi -- I'm going to do a deeper review of this shortly. Could you update the `changelog.md` file and list all of the changes we've made?  It's nice to have that for when we do a release and want to provide release notes.

File: tests/convergence_tests/check_for_meshes.py (hunk 1 above)
Comment (ncclementi) [inline tests/convergence_tests/check_for_meshes.py]: All fixed, thanks @gforsyth

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,24 @@
 
 * Ported PyGBe to Python 3 (!!!).  This breaks Python 2 support, but who cares.
 * Better regression tests (faster, anyway) using pytest
+* Localized Surface Plasmon Resonance application:
+
+    #### Added:
```
Comment (ncclementi) [inline CHANGELOG.md]: I wasn't sure how to indicate that those additions and changes are related to the lspr application, but I guess yes we can, put in the same level.

File: pygbe/lspr.py [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+import os
+import re
+import sys
+import time
+import glob
+import numpy
+import pickle
+import subprocess
+from datetime import datetime
+from argparse import ArgumentParser
```
Comment (ncclementi) [inline pygbe/lspr.py]: We import the `read_inputs` from `pygbe.main` and we use that. I don't see why we should do a different one...

File: pygbe/lspr.py [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+This is the main function of the program.
+We use a boundary element method (BEM) to perform simple nanoparticle
+plasmonics. Localized surface plasmon resonance (LSPR) is an optical effect,
+but electrostatics is a good approximation in the long-wavelength limit. For
+nanoparticles smaller than the wavelength of incident light, PyGBe 
+compute the extinction cross-section.
+"""
+
+import os
+import re
```
Comment (ncclementi) [inline pygbe/lspr.py]: we do use `os`, line 73, 77, 79 and so more.

File: pygbe/lspr.py [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+import sys
+import time
+import glob
```
Comment (ncclementi) [inline pygbe/lspr.py]: we also use `sys` and `time`,

File: pygbe/lspr.py (hunk 4 above)
Comment (gforsyth) [inline pygbe/lspr.py]: Ahh, ok, well then we should remove this import

File: pygbe/lspr.py (hunk 5 above)
Comment (gforsyth) [inline pygbe/lspr.py]: sorry, the comments refer only to the bottom-most line. `re` is unused

File: pygbe/lspr.py (hunk 6 above)
Comment (gforsyth) [inline pygbe/lspr.py]: `glob` is unused

File: pygbe/lspr.py (hunk 4 above)
Comment (ncclementi) [inline pygbe/lspr.py]: Sure, I mean we are importing just the function from `pygbe.main` not the module. Will it find it if we don't added in the `lspr.py` script.?

Comment (gforsyth) [inline pygbe/lspr.py]: Yes

File: pygbe/lspr.py [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+    for i in range(len(surf_array)):
+        time_sort += surf_array[i].fill_surface(param)
+
+    ### Output setup summary
+    param.N = 0
+    param.Neq = 0
+    for s in surf_array:
+        N_aux = len(s.triangle)
+        param.N += N_aux
+        if s.surf_type == 'dirichlet_surface' or s.surf_type == 'neumann_surface' or s.surf_type == 'asc_surface':
```
Comment (ncclementi) [inline pygbe/lspr.py]: Ok sounds good, I'll change it in `pygbe.main` too.

**Comment by gforsyth** [comment]
Looks good, @ncclementi !  Nice work!

**Pull request merged**